### ISSUE TITLE
SQLObject instance objects unhashable on python 3

### ISSUE THREAD

**Issue by drnlm** [sqlobject/sqlobject#115]
While porting a local application to Python 3 using SQLObject 3, I ran into the following corner case

```python
from sqlobject import SQLObject, UnicodeCol

class A(SQLObject):
    name = UnicodeCol()

sqlhub.processConnection = connectionForURI('sqlite:///:memory:')
A.createTable()
x=A(name='test1')
d={x: 'lookup'}
```

This works fine with python 2, but fails using python 3 with ```TypeError: unhashable type: 'A'```. This is because SQLObject defines ```__eq__```, and in python 3, this disables the implicit ```object.__hash__``` implementation

While in the case I'm porting, it's easy enough to change the code to work around this issue, there are probably other people depending on this behaviour, so it's worthwhile having some official fix.

As I see it, there are 3 possible approaches to this issue:

1. Do nothing and documention this change. In python 2, this only works becuase of the default id()-based implementation, and hash values are consequently not stable across cache expiry and probably some other cases.

2. Revert to the python 2 behaviour by explicitly enabling the default ```__hash__``` implementation. According to docs.python.org, adding something like ```__hash__ = object.__hash__``` to the class definition will do this.

3. Provide a ```__hash__``` implementation for SQLObject that is stable across cache expiry. Something based on the class name and the database id is probably good enough for this.

**Comment by phdru** [comment]
I vote for the option 3. Exactly that — class name + id.

**Comment by phdru** [comment]
Fixed in #117.
